### DISCUSSION TITLE
Create an objectProperty with an expression as the key

### DISCUSSION THREAD

**Discussion by gabrielschulhof** [babel/babel#16433] [Q&A]
I am doing a transform where I copy the list of ObjectProperty nodes from an objectExpression to another:

```JS
some.deep.tree.arguments[0] = t.objectExpression(existingObjectExpression.properties)
```

This works fine if all properties of `existingObjectExpression.properties` have literals for keys, but the resulting transform fails if the key for one of the `ObjectProperty` items is, say, a `ConditionalExpression`. Do you know how I can get the transformation to work with `ConditionalExpression` as the `ObjectProperty`s key?

**Comment by liuxingbaoyu** [comment]
Can you provide a code example?

**Comment by gabrielschulhof** [comment]
@liuxingbaoyu as I was building the code example, I must've made some change to make it work. So, the code example has now become the solution I was trying to build in the first place 😄 Thanks for prompting me 🙂
